src/expr/skolem_manager.h

```diff
@@ -149,13 +169,35 @@ class SkolemManager
    * Mapping from witness terms to proof generators.
    */
   std::map<Node, ProofGenerator*> d_gens;
+  /** Map to canonical bound variables */
+  std::map<std::pair<Node, Node>, Node> d_witnessBoundVar;
   /** Convert to witness or skolem form */
   static Node convertInternal(Node n, bool toWitness);
   /** Get or make skolem attribute for witness term w */
   static Node getOrMakeSkolem(Node w,
                               const std::string& prefix,
                               const std::string& comment,
                               int flags);
+  /**
+   * Skolemize the first variable of existentially quantified formula q.
+   * For example, calling this method on:
+   *   (exists ((x Int) (y Int)) (P x y))
+   * will return:
+   *   (witness ((x Int)) (exists ((y Int)) (P x y)))
+   * If q is not an existentially quantified formula, then null is
+   * returned and an assertion failure is thrown.
+   *
+   * This method additionally updates qskolem to be the skolemized form of q.
+   * In the above example, this is set to:
+   *   (exists ((y Int)) (P (witness ((x Int)) (exists ((y Int)) (P x y))) y))
+   */
+  Node skolemize(Node q,
+                 Node& qskolem,
+                 const std::string& prefix,
+                 const std::string& comment = "",
+                 int flags = NodeManager::SKOLEM_DEFAULT);
+  /** Get or make bound variable */
```

Comment:
say this uses the canonical mapping above to ensure uniqueness up to `t,s`. Also clarify what should `t` and `s` be.